azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/ConfigMojo.java

```diff
@@ -47,201 +53,220 @@
             throw new MojoFailureException(e.getMessage());
         } finally {
-            try {
-                this.prompt.close();
-            } catch (IOException e) {
-                // ignore at final step
-            }
-        }
-    }
-
-    private void initializeProjects() throws MojoFailureException, IOException {
-        final List<MavenProject> allProjects = session.getAllProjects().stream().filter(Utils::isJarPackagingProject).collect(Collectors.toList());
-        final List<String> configuredProjects = new ArrayList<>();
-        for (final MavenProject proj : allProjects) {
-            if (isProjectConfigured(proj)) {
-                configuredProjects.add(proj.getName());
-            } else {
-                targetProjects.add(proj);
+            if (this.wrapper != null) {
+                try {
+                    this.wrapper.close();
+                } catch (IOException e) {
+                    // ignore at final step
+                }
             }
-        }
 
-        if (targetProjects.size() == 0) {
-            getLog().warn("There are no projects in current folder which are not configured yet!");
-            return;
         }
+    }
 
-        if (configuredProjects.size() > 0) {
-            getLog().warn(String.format("Project (%s) %s already configured.",
-                    TextUtils.yellow(configuredProjects.stream().collect(Collectors.joining(", "))), configuredProjects.size() > 1 ? "are" : "is"));
-        }
+    private void configCommon() throws IOException, ExpressionEvaluationException, NoResourcesAvailableException, InvalidConfigurationException {
+        configureAppName();
+        configurePublic();
+        configureInstanceCount();
+        configureCpu();
+        configureMemory();
+        configureJvmOptions();
+    }
 
-        if (targetProjects.size() == 1) {
-            if (!prompt.promoteYesNo(true,
-                    String.format("There is only one project(%s) which is not configured, would you like to select it to config?",
-                            TextUtils.blue(targetProjects.get(0).getName())), true)) {
-                // user cancels
-                targetProjects.clear();
+    private void selectProjects() throws MojoFailureException, IOException, NoResourcesAvailableException {
+        if (this.parentMode) {
+            final List<MavenProject> allProjects = session.getAllProjects().stream().filter(Utils::isJarPackagingProject).collect(Collectors.toList());
+            final List<MavenProject> configuredProjects = new ArrayList<>();
+            for (final MavenProject proj : allProjects) {
+                if (isProjectConfigured(proj)) {
+                    configuredProjects.add(proj);
+                } else {
+                    targetProjects.add(proj);
+                }
             }
-            return;
+            this.wrapper.putCommonVariable("projects", targetProjects);
```

Comment:
Dup with last line, remove this one?

Reply:
No, the first time, set all available projects for rendering the project list for user to select, on the second step, the target projects will be the project list selected by user